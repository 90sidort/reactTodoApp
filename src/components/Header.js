import React from 'react';
import { NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Todo app</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact={true}>Create todo</NavLink>
        <NavLink to="/help" activeClassName="is-active" exact={true}>Help</NavLink>
    </header>
)

export default Header