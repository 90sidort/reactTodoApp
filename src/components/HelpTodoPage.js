import React from 'react';

const HelpTodoPage = () => (
    <div>
        This is my todo help page.
    </div>
)

export default HelpTodoPage