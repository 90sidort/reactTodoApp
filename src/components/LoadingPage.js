import React from "react";

const LoadingPage = () => (
  <div className="loader">
    <img src="/images/loader.gif" />
    <p>Loading...</p>
  </div>
);

export default LoadingPage;
