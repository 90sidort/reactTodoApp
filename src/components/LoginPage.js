import React from "react";
import { connect } from "react-redux";
import { startLogin, startLogUser } from "../actions/auth";

export const LoginPage = ({ startLogin, startLogUser }) => {
  return (
    <div className="box-layout">
      <div className="box-layout__box">
        <h1 className="box-layout__title">Todo App</h1>
        <p>It's time to get more organized!</p>
        <button className="buttonBlue" onClick={startLogin}>
          Login with Google
        </button>
        <button className="buttonBlue" onClick={startLogUser}>
          Test user
        </button>
      </div>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startLogUser: () => dispatch(startLogUser()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
