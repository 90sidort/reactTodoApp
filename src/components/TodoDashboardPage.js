import React from 'react';
import ConnectedTodoList from './TodoList'
import TodoListFilters from '../components/TodoListFilters'

const TodoDashboardPage = () => (
    <div>
        This is my TodoDashboardPage.
        <TodoListFilters/>
        <ConnectedTodoList/>
    </div>
)

export default TodoDashboardPage