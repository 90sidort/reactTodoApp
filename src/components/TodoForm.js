import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

class TodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: props.todo ? props.todo.title : "",
      completed: props.todo ? props.todo.completed : false,
      details: props.todo ? props.todo.details : "",
      priority: props.todo ? props.todo.priority : 1,
      createdAt: props.todo ? moment(props.todo.createdAt) : moment(),
      calendarFocused: false,
      error: "",
    };
  }
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };
  onDetailsChange = (e) => {
    const details = e.target.value;
    this.setState(() => ({ details }));
  };
  onPriorityChange = (e) => {
    const priority = e.target.value;
    this.setState(() => ({ priority }));
  };
  onDateChange = (createdAt) => {
    this.setState(() => ({ createdAt }));
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.title) {
      this.setState({ error: "Please, provide title" });
    } else {
      this.setState({ error: "" });
      this.props.onSubmit({
        title: this.state.title,
        completed: this.state.completed,
        priority: parseInt(this.state.priority),
        createdAt: this.state.createdAt.valueOf(),
        details: this.state.details,
      });
    }
  };
  onCompletedChange = (e) => {
    e.preventDefault();
    if (!this.state.completed) {
      this.setState(() => ({ completed: true }));
    } else {
      this.setState(() => ({ completed: false }));
    }
  };
  render() {
    let buttonText;
    if (!this.state.completed) {
      buttonText = `Mark as done`;
    } else {
      buttonText = `Mark as undone`;
    }
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            value={this.state.title}
            placeholder="Todo title"
            onChange={this.onTitleChange}
          />
          <button onClick={this.onCompletedChange}>{buttonText}</button>
          <select value={this.state.priority} onChange={this.onPriorityChange}>
            <option value={-1} disabled>
              Select priority:
            </option>
            <option value={0}>Low</option>
            <option value={1}>Normal</option>
            <option value={2}>High</option>
          </select>
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            placeholder="Provide description (optional)"
            value={this.state.details}
            onChange={this.onDetailsChange}
          />
          ]<button>Create todo</button>
        </form>
      </div>
    );
  }
}

export default TodoForm;
