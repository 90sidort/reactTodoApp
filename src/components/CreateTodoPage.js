import React from "react";
import TodoForm from "./TodoForm";
import { connect } from "react-redux";
import { startAddTodo } from "../actions/todos";

export class CreateTodoPage extends React.Component {
  onSubmit = (todo) => {
    this.props.startAddTodo(todo);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <div className="pageHeader">
          <div className="summaryContent">
            <h1 className="pageHeaderTitle">Create Todo</h1>
          </div>
        </div>
        <div className="content-container">
          <TodoForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddTodo: (todo) => dispatch(startAddTodo(todo)),
});

export default connect(undefined, mapDispatchToProps)(CreateTodoPage);
