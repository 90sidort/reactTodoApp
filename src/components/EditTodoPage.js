import React from 'react';
import { connect } from 'react-redux'
import TodoForm from './TodoForm'
import { editTodo, removeTodo } from '../actions/todos'

export class EditTodoPage extends React.Component {
    onSubmit = (todo) => {
        this.props.editTodo(this.props.todo.id, todo)
        this.props.history.push('/')
    }
    onRemove = () => {
        this.props.removeTodo({id: this.props.todo.id})
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <TodoForm 
                    todo={this.props.todo}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onRemove}
                >
                    Remove
                </button>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        todo: state.todos.find((todo) => todo.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch) => ({
    editTodo: (id, todo) => dispatch(editTodo(id, todo)),
    removeTodo: (data) => dispatch(removeTodo(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditTodoPage)