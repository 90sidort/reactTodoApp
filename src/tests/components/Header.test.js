import { Header } from "../../components/Header";
import React from "react";
import { shallow } from "enzyme";

let startLogout = jest.fn();

test("Should correctly render Header component", () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test("Should call startLogout on button click", () => {
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find("button").simulate("click");
  expect(startLogout).toHaveBeenCalled();
});
