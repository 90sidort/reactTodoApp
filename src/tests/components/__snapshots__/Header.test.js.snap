// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Header component 1`] = `
<header
  className="header"
>
  <div
    className="headerContent"
  >
    <div
      className="headerMenu"
    >
      <h1
        className="header__title"
      >
        Todo app
      </h1>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="linkElement"
        exact={true}
        to="/dashboard"
      >
        Dashboard
      </NavLink>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="linkElement"
        exact={true}
        to="/create"
      >
        Create todo
      </NavLink>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="linkElement"
        exact={true}
        to="/help"
      >
        Help
      </NavLink>
    </div>
    <button
      className="buttonLogout"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
